File: php/features.php
<?php
declare(strict_types=1);

// AiCats SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class AiCatsFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new AiCatsBaseFeature();
            case "test":
                return new AiCatsTestFeature();
            default:
                return new AiCatsBaseFeature();
        }
    }
}

File: py/features.py
# AiCats SDK feature factory

from feature.base_feature import AiCatsBaseFeature
from feature.test_feature import AiCatsTestFeature


def _make_feature(name):
    features = {
        "base": lambda: AiCatsBaseFeature(),
        "test": lambda: AiCatsTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
